File: uHunt/C++src/easy/RequestForProposal.cpp
using namespace std;
#include <iostream>
#include <string>
#include <vector>

struct propuesta
{
    string nombre;
    double precio;
    int numReqCumlplidos;
};

int main()
{
    int numRequisitos;cin >> numRequisitos;
    int numEmpresas;cin >> numEmpresas;
    for(int i=0; numEmpresas!=0 && numRequisitos!=0; i++)
    {
        string req;
        for(int j=0; j<numRequisitos; j++) getline(cin,req);

        vector<propuesta> props;
        int maxReqsCumplidos=0;
        for (int j = 0; j < numEmpresas; j++)
        {
            props.push_back(propuesta());
            cin >> props[j].nombre;
            cin >> props[j].precio;
            cin >> props[j].numReqCumlplidos;

            int auxl=props[j].numReqsCumplidos;
            if(props[j].numReqCumlplidos>maxReqsCumplidos)maxReqsCumplidos=props[j].numReqCumlplidos;}

            string reqcumple;
            for (int k = 0; k < auxl ; k++)getline(cin,reqcumple);
		}

		vector<propuesta> primerFiltro;
		double menorPrecio=-1.0;
		int tamSegFiltro=0;
		for (int j = 0; j < numEmpresas; j++)
        {
            if(props[j].numReqCumlplidos==maxReqsCumplidos)
            {
                primerFiltro.push_back(props[j]);
                if(props[j].precio<menorPrecio || menorPrecio==-1)menorPrecio=props[j].precio;
                tamSegFiltro++;
            }
        }

        vector<propuesta> segundoFiltro;
        for (int j = 0; j < tamSegFiltro; j++) if(primerFiltro[j].precio==menorPrecio)segundoFiltro.push_back(primerFiltro[j]);

        cout << "RFP #" << i << endl;
        cout << segundoFiltro[0] << endl;

        cin >> numRequisitos;
        cin >> numEmpresas;
    }
    return 0;
}

File: uHunt/src/easy/RequestForProposal.java
package easy;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;

public class RequestForProposal 
{
	private static BufferedReader br;
	private static boolean test=true;

	public static void main(String[] args) throws IOException 
	{
		if(test)br=new BufferedReader(new FileReader(new File("./data/easy/RequestForProposal.in")));
		else br=new BufferedReader(new InputStreamReader(System.in));
		solucinarProblema();
		br.close();
	}

	private static void solucinarProblema() throws IOException
	{
		String linea=br.readLine(), empresa;
		String[] param, paramPropuesta;
		int numRequisitos, numEmpresas, maxReqsCumplidos, numReqsCumplidos;
		double precio;
		ArrayList<Propuesta> primerFiltro, segundoFiltro;
		Double menorPrecio;
		
		for(int i=1; !linea.equals("0 0"); i++)
		{
			if(i>1) System.out.println();

			param=linea.split(" ");
			numRequisitos=Integer.parseInt(param[0]);
			numEmpresas=Integer.parseInt(param[1]);

			for (int j=0; j<numRequisitos; j++)br.readLine();

			ArrayList<Propuesta> proEmpresas=new ArrayList<Propuesta>();
			maxReqsCumplidos=0;
			for (int j=0; j<numEmpresas; j++)
			{
				empresa=br.readLine();
				paramPropuesta=br.readLine().split(" ");
				precio=Double.parseDouble(paramPropuesta[0]);
				numReqsCumplidos=Integer.parseInt(paramPropuesta[1]);

				if(numReqsCumplidos>maxReqsCumplidos) maxReqsCumplidos=numReqsCumplidos;

				for (int k=0; k<numReqsCumplidos; k++)br.readLine();

				proEmpresas.add(new Propuesta(empresa,precio,numReqsCumplidos));
			}

			primerFiltro=new ArrayList<Propuesta>();
			menorPrecio=Double.POSITIVE_INFINITY;
			
			for (Propuesta propuesta : proEmpresas)
			{
				if(propuesta.numReqsCumplidos==maxReqsCumplidos)
				{
					primerFiltro.add(propuesta);
					if(propuesta.precio<menorPrecio) menorPrecio=propuesta.precio;
				}
			}

			segundoFiltro=new ArrayList<Propuesta>();
			for (Propuesta propuesta : primerFiltro)
				if(propuesta.precio==menorPrecio) segundoFiltro.add(propuesta);

			System.out.println("RFP #"+i);
			System.out.println(segundoFiltro.get(0).empresa);
			linea=br.readLine();
		}
	}

	private static class Propuesta
	{
		String empresa;
		Double precio;
		int numReqsCumplidos;
		
		public Propuesta(String empresaP, Double precioP, int numReqsCumplidosP)
		{
			empresa=empresaP;
			precio=precioP;
			numReqsCumplidos=numReqsCumplidosP;
		}
	}
	
}